packages/regenesis-surgery/scripts/surgery.ts

```diff
@@ -1,36 +1,37 @@
-import {
-  StateDump,
-  UniswapPoolData,
-  SurgeryDataSources,
-  EtherscanContract,
-} from './types'
+import { StateDump, SurgeryDataSources } from './types'
 import { handlers } from './handlers'
 import { classify } from './classifiers'
+import { findAccount } from './utils'
 
-const main = async () => {
-  const dump: StateDump = null as any // TODO
-  const genesis: StateDump = null as any // TODO
-  const pools: UniswapPoolData[] = null as any // TODO
-  const etherscanDump: EtherscanContract[] = null as any // TODO
-  const data: SurgeryDataSources = {
-    dump,
-    genesis,
-    pools,
-    etherscanDump,
-    l1TestnetProvider: null as any, // TODO
-    l1MainnetProvider: null as any, // TODO
-    l2Provider: null as any, // TODO
-  }
-
-  // TODO: Insert any accounts from genesis that aren't in the dump
-
+const doGenesisSurgery = async (
+  data: SurgeryDataSources
+): Promise<StateDump> => {
+  // We'll generate the final genesis file from this output.
   const output: StateDump = []
-  for (const account of dump) {
+
+  // Handle each account in the state dump.
+  for (const account of data.dump) {
     const accountType = classify(account, data)
     const handler = handlers[accountType]
-    const newAccount = await handler(account, data)
+    const newAccount = await handler(JSON.parse(JSON.stringify(account)), data)
     if (newAccount !== undefined) {
       output.push(newAccount)
     }
   }
+
+  // Injest any accounts in the genesis that aren't already in the state dump.
+  for (const account of data.genesis) {
+    if (findAccount(data.dump, account.address) === undefined) {
+      output.push(account)
+    }
+  }
+
+  return output
+}
+
+const main = async () => {
+  // TODO
```

Comment:
assuming this is where populating the `SurgeryDataSources` blob happens? like what i add in https://github.com/ethereum-optimism/optimism/pull/1530?

Reply:
Correct!